src/mgr/PyModules.cc

```diff
@@ -544,8 +544,13 @@ void PyModules::set_config(const std::string &handle,
   }
   set_cmd.wait();
 
-  // FIXME: is config-key put ever allowed to fail?
-  assert(set_cmd.r == 0);
+  if (set_cmd.r != 0) {
+    // config-key put will fail if mgr's auth key has insufficient
+    // permission to set config keys
+    // FIXME: should this somehow raise an exception back into Python land?
+    dout(0) << "`config-key put " << global_key << " " << val << "` failed: "
+      << cpp_strerror(set_cmd.r) << dendl;
```

Comment:
I just tried adding `dout(0) << set_cmd.outs << dendl;` but it doesn't seem to add much useful. The existing code prints:

````
2017-04-21 22:06:30.807527 7f1fdfd60700  0 mgr set_config `config-key put mgr.re
st.enable_auth false` failed: (13) Permission denied
````

If I add `set_cmd.outs` we also get:

````
2017-04-21 22:06:30.807538 7f1fdfd60700  0 mgr set_config access denied
````